AirCasting/CoreData/SessionEntity.swift

```diff
@@ -39,6 +38,9 @@ public class SessionEntity: NSManagedObject, Identifiable {
     @NSManaged public var changesCount: Int32
     @NSManaged public var rowOrder: Int64
     
+    // UserInterface state
+    @NSManaged public var userInterface: UIStateEntity?
```

Comment:
I'm thinking whether it wouldn't be better not to add this relationship to session entity but rather add SessionUUID property to UIStateEntity and operate on that. What do you think about that? This way we wouldn't have to add this UIState logic to Session logic and just keep it in the session card view for followed sessions

Reply:
I need to say I would like to leave it like it is if you agree. The reasons are:
- now it is well connected with a relation 1:1 and everything is clear
- whenever we would need to find a proper UIState (when using UUID) we would need to iterate and find a proper one then. I think it is not needed
- The pros is whenever we delete the session card we get rid of the UIState too. In the UUID way, we would need to iterate and delete it separately

What do you think?

Reply:
I agree with @jankrzempek here. On CoreData level this tradeoff seems really beneficial - pros outweigh cons. As long as we keep it well separated on the interface level so the rest of the app see the ui state database and core database as separate modules.